src/goto-instrument/contracts/assigns.cpp

```diff
@@ -89,17 +88,18 @@ assigns_clauset::assigns_clauset(
   const exprt &expr,
   const messaget &log,
   const namespacet &ns)
-  : expr(expr), log(log), ns(ns)
+  : location(expr.source_location()), log(log), ns(ns)
 {
   for(const auto &target_expr : expr.operands())
   {
-    add_target(target_expr);
+    add_global_write_set(target_expr);
   }
 }
 
-void assigns_clauset::add_target(const exprt &target_expr)
+void assigns_clauset::add_global_write_set(const exprt &target_expr)
```

Comment:
[nitpick]

I liked the names we discussed, `add_to_global_write_set` and `remove_from_global_write_set` better. The current names are a bit confusing because it sounds like we are adding a whole set or removing a whole set.

Before it was `add_target` and that made sense, because we were adding a new target.

Reply:
Implemented as suggested.